BlueprintUI/Sources/Layout/Builder/StackElement+Builder.swift

```diff
@@ -0,0 +1,115 @@
+        self.traits = traits
+        self.key = key
+    }
+    
+    public init(
+        wrappedElement: Element,
+        growPriority: CGFloat = 1.0,
+        shrinkPriority: CGFloat = 1.0,
+        alignmentGuide: ((ElementDimensions) -> CGFloat)? = nil,
+        key: AnyHashable? = nil
+    ) {
+        self.init(
+            wrappedElement: wrappedElement,
+            traits: .init(
+                growPriority: growPriority,
+                shrinkPriority: shrinkPriority,
+                alignmentGuide: alignmentGuide.map(StackLayout.AlignmentGuide.init)
+            ),
+            key: key
+        )
+    }
+
+    // Simply wraps the given element.
+    public var elementRepresentation: Element { wrappedElement }
+}
+
+extension Element {
+    
+    /// Wraps an element with a `StackChild` in order to customize `StackLayout.Traits` and the key.
+    /// - Parameters:
+    ///   - growPriority: Controls the amount of extra space distributed to this child during underflow.
+    ///   - shrinkPriority: Controls the amount of space allowed for this child during overflow.
+    ///   - alignmentGuide: Allows for custom alignment of a child along the cross axis.
+    ///   - key: A key used to disambiguate children between subsequent updates of the view
+    ///     hierarchy.
+    /// - Returns: A wrapped element with additional layout information for the `StackElement`.
+    public func stackChild(
+        growPriority: CGFloat = 1.0,
+        shrinkPriority: CGFloat = 1.0,
+        alignmentGuide: ((ElementDimensions) -> CGFloat)? = nil,
+        key: AnyHashable? = nil
+    ) -> StackChild {
+        .init(
+            wrappedElement: self,
+            growPriority: growPriority,
+            shrinkPriority: shrinkPriority,
+            alignmentGuide: alignmentGuide, key: key
+        )
+    }
+    
+    /// Wraps an element with a `StackChild` in order to customize `StackLayout.Traits` and the key.
+    /// - Parameters:
+    ///   - traits: Contains traits that affect the layout of individual children in the stack.
+    ///   - key: A key used to disambiguate children between subsequent updates of the view
+    ///     hierarchy.
+    /// - Returns: A wrapped element with additional layout information for the `StackElement`.
+    public func stackChild(traits: StackLayout.Traits = .init(), key: AnyHashable? = nil) -> StackChild {
+        .init(wrappedElement: self, traits: traits, key: key)
+    }
+}
+
+
+extension StackElement {
+    /// Initializer using result builder to declaritively build up a stack.
+    /// - Parameters:
+    ///   - children: A block containing all elements to be included in the stack.
+    ///   - configure: A closure used to modify the stack.
+    /// - Note: If element is a StackChild, then traits and key will be pulled from the element, otherwise defaults are passed through.
+    public init(@Builder<Element> elements: () -> [Element], configure: (inout Self) -> Void = { _ in }) {
```

Comment:
I'd put the `configure` block first, since it's usually a lot shorter – so harder to lose after lots of content building. Thoughts?

Reply:
I started with the configure block first, but switched to make it second for a couple reasons:
1. Consistency - the other elements I have used all seem to place the configure block last
2. When I am building a layout I start with building the hierarchy and then tweaking the containers and elements. Like this:

```swift
// 1)
Column {
    Element1()
    Text("Hello, world!")
    Element2()
}

// 2)
Column {
    Element1()
    Text("Hello, world!") { 
        $0.lineLimit = 1
    }
    Element2()
} configure: {
    $0.horizontalUnderflow(.center)
}
```

By having the configure block come second I can simply append any modifications rather whereas if I were to reverse the parameters then I would need to insert the `elements` label into the code. Its nothing major but is (imo) a small quality of life improvement.